src/components/TeamDetails.jsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import axios from 'axios';
+import BackButton from './BackButton';
+
+const TeamDetails = () => {
+  const { id } = useParams();
+  const [gameData, setGameData] = useState(null);
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const response = await axios.get(
+          `https://www.balldontlie.io/api/v1/games/${id}`,
+        );
+        setGameData(response.data);
+      } catch (error) {
+        throw new Error('an exception has occured');
+      }
+    }
+
+    fetchData();
```

Comment:
- [ OPTIONAL ] Good job working on the details page as well but I think instead of making the functionality of the API here you could make a slice in redux folder and add it there with the method and that's the functionality of redux to handle the API but you can just call the method throughout the store 🙏🏻🔥🎉

- [ OPTIONAL ] I would suggest for you to add store file in the redux folder to have structured codebase ✅